ecl/hql/hqlgram.y

```diff
@@ -8812,6 +8812,10 @@ simpleDataSet
 
                             bool hasFileposition = getBoolAttributeInList(extra, filepositionAtom, true);
                             record.setown(parser->checkIndexRecord(record, $5, extra));
+
+                            if (!parser->checkAllowed($1, "datafile", "Datafile access"))
+                                extra.setown(createComma(extra.getClear(), createAttribute(_disallowed_Atom)));
```

Comment:
From my side yes.
It wasn't (still isn't) obvious why done like this, but it is not new, i.e. it's the same pattern as e.g. PIPE).